ad/movement/ad-cs/certificate-templates.md

```diff
@@ -535,6 +585,222 @@ The certificate can then be used with [Pass-the-Certificate](../kerberos/pass-th
 {% endtab %}
 {% endtabs %}
 
+### Weak explicit mapping (ESC14)
+
+This attack exploits configuration errors when **explicit** mapping is set up. To understand this ESC, it is essential to have a good command of [Certificate mapping](certificate-templates.md#certificate-mapping).
+
+There are four possible attack scenarios for exploiting ESC14. In all cases, the following prerequisites must be met:
+
+{% hint style="warning" %}
+* the attacker has compromised a **victim** account and is able to request certificates with this account, and wants to compromise a **target** account
+* the certificate template allows the **victim** to enrol
+* the **victim** matches all the prerequisites for issuing the certificate
+* the certificate template permits authentication
+* if the template has the value `CT_FLAG_SUBJECT_ALT_REQUIRE_UPN` or `CT_FLAG_SUBJECT_ALT_REQUIRE_SPN` in the `msPKI-Certificate-Name-Flag` attribute, then:
+  * The `UseSubjectAltName` registry key on the DC must be set to `0`
+  * authentication can only be performed via PKINIT (Kerberos)
+* if the template indicates `CT_FLAG_SUBJECT_ALT_REQUIRE_DNS` or `CT_FLAG_SUBJECT_ALT_REQUIRE_DOMAIN_DNS` in the `msPKI-Certificate-Name-Flag` attribute:
+  * the **victim** must be a machine
+  * authentication can only be performed via PKINIT (Kerberos)
+* if the template indicates `CT_FLAG_SUBJECT_ALT_REQUIRE_EMAIL` or `CT_FLAG_SUBJECT_REQUIRE_EMAIL` in the `msPKI-Certificate-Name-Flag` attribute, one of the following prerequisites must be validated:
+  * the certificate template uses version 1 of the scheme
+  * the **victim** has his `mail` attribute configured
+  * the attacker has write access to the **victim**'s `mail` attribute
+{% endhint %}
+
+Sufficient DACL to write `altSecurityIdentities` attributes can be detected using the [Get-WriteAltSecIDACEs.ps1](https://github.com/JonasBK/Powershell/blob/master/Get-WriteAltSecIDACEs.ps1) script (PowerShell).
+
+Detection of weak explicit mapping can be done with the [Get-AltSecIDMapping.ps1](https://github.com/JonasBK/Powershell/blob/master/Get-AltSecIDMapping.ps1) script (PowerShell).
```

Comment:
could dacledit/bloodhound or any other cross-platform/UNIX-like tool be used here?